packages/relay/src/lib/services/lockService/LocalLockStrategy.ts

```diff
@@ -0,0 +1,162 @@
+// SPDX-License-Identifier: Apache-2.0
+
+import { Mutex } from 'async-mutex';
+import { randomUUID } from 'crypto';
+import { LRUCache } from 'lru-cache';
+import { Logger } from 'pino';
+
+/**
+ * Represents the internal state for a lock associated with a given address.
+ */
+interface LockState {
+  mutex: Mutex;
+  sessionKey: string | null;
+  acquiredAt: number | null;
+  maxLockTime: NodeJS.Timeout | null;
```

Comment:
Should we rename this as lockTimeOutId for better clarity? It holds the id so we can remove it later, right?

```suggestion
lockTimeoutId: NodeJS.Timeout | null;
```